### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by estimpson** [microsoft/CsWin32#345] [Q&A] [answered]
I'm using VS 2019 (Community) with a new Console Application [not Console App (.NET Framework)].  I add the nuget package, create the NativeMethods.txt file in the project's root folder, add  a single line with "CreateFile", and press F6 to build.  When I try to add a using for Microsoft.Windows.Sdk to my Program.cs, all I get is the "type or namespace 'Windows' does not exist..."

I verified that I have the .NET 5.0 SDK installed:
![image](https://user-images.githubusercontent.com/3619926/127692410-785f8ff4-7db5-45db-93bc-6bffb88d91ec.png)

And I'm running version 16.10.1 of VS.

Am I missing a step somewhere?  Or can someone create a minimal GitHub repository with a working console App that I can clone to figure out what I'm doing wrong?

**Comment by jnm2** [comment, marked as answer]
`Microsoft.Windows.Sdk` is no longer the namespace. Is that being displayed somewhere that you're copying from?

(https://github.com/microsoft/CsWin32/issues/326)

**Reply by estimpson** [reply]
That's right from the readme:
![image](https://user-images.githubusercontent.com/3619926/127695254-9ff5c738-c5cd-491e-a530-19c201df9693.png)

What's it supposed to be?

**Reply by estimpson** [reply]
So I see the issue you linked, and I changed my using statement to:
````using Windows.Win32;````
But now it's just saying that "The type or namespace name 'Windows' could not be found..."

**Reply by jnm2** [reply]
It's working for me. Sometimes VS2019 is slow to catch up. Also make sure this is not coming from ReSharper which is even slower to catch up (requires VS reboot IIRC).

**Reply by estimpson** [reply]
Any way you could create a repo with the CreateFile example working?

**Reply by estimpson** [reply]
I had better luck with VS Code, so maybe it is just ReSharper, which I am using

**Reply by jnm2** [reply]
Here's a repo using `0.1.506-beta` and other APIs: https://github.com/Techsola/InstantReplay. I know CreateFile works as well as these APIs because we use it in a private repo.

Sometimes updating ReSharper and VS can make things like this happen less often. I'm using VS 16.10.4 and ReSharper 2021.1.3 and I have to build the project/solution and then wait a few seconds for ReSharper to see the generated code. Then all is fine.

**Reply by estimpson** [reply]
Thank you!  I'll check it out.

And thanks for the tips on ReSharper.  I'm up to date on both, but I might have some suboptimal configurations in ReSharper.  I'll see if I can figure out what's going on and if I do I'll post back

**Reply by jnm2** [reply]
I noticed you weren't up to date as of the beginning of the thread (you mentioned VS 16.10.1) so I thought it was especially worth mentioning before I went all "works on my machine." 😆
